### PR TITLE
KAFKA-5733: RocksDB bulk load with lower number of levels.

### PR DESCRIPTION
**Pull request by guozhangwang**
This is to complete Bill's PR #3664 on KAFKA-5733, incorporating the suggestion in https://github.com/facebook/rocksdb/issues/2734.

Some minor changes: move `open = true` in `openDB`.

### CODE REVIEW

**Comment by guozhangwang** [comment]
@dguy @bbejeck for reviews.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6835/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6821/
Test PASSed (JDK 8 and Scala 2.12).

File: streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java [hunk 1]
```diff
@@ -254,10 +256,34 @@ private void validateStoreOpen() {
...
+            if (sstFileNames != null && sstFileNames.length > 0) {
+                try {
+                    this.db.compactRange(true, 1, 0);
+                } catch (RocksDBException e) {
+                    throw new ProcessorStateException("Error while range compacting during restoring  store " + this.name, e);
+                }
+
+                // we need to re-open with the old num.levels again, this is a workaround
+                // until https://github.com/facebook/rocksdb/pull/2740 is merged in rocksdb
+                close();
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:279]: Do we need to do this `close` and `open` here? We do it also on lines 283 & 286

**Review by dguy — COMMENTED** [review]
Just a question around closing and opening the store inside the if block, otherwise LGTM

File: streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:279]: Unfortunately it is needed as suggested here: https://github.com/facebook/rocksdb/issues/2734

**Comment by dguy** [comment]
LGTM

**Comment by guozhangwang** [comment]
Merged to trunk

**Pull request closed without merging**